shell/platform/fuchsia/dart_runner/parse_url.cc

```diff
@@ -0,0 +1,30 @@
+#include "parse_url.h"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <string>
+
+namespace dart_runner {
+
+std::string GetLabelFromUrl(const std::string& url) {
+  for (size_t i = url.length() - 1; i > 0; i--) {
+    if (url[i] == '/') {
+      return url.substr(i + 1, url.length() - 1);
+    }
+  }
+  return url;
+}
+
+// Find the name of the component.
+// fuchsia-pkg://fuchsia.com/hello_dart#meta/hello_dart.cm -> hello_dart
```

Comment:
One note: this is something I added in my original PR since it matches how we build flutter applications in tree right now. In the future we will probably want to avoid making this assumption and only use it as a default. We should define an API for the cml file which allows users to set this value directly and then pull that out from the program dictionary in the startup info https://cs.opensource.google/fuchsia/fuchsia/+/main:sdk/fidl/fuchsia.component.runner/component_runner.fidl;l=64.